test/groovy/TransportRequestUploadFileTest.groovy

```diff
@@ -0,0 +1,114 @@
+            binding.setProperty('username', 'anonymous')
+            binding.setProperty('password', '********')
+            try {
+                c()
+            } finally {
+                binding.setProperty('username', null)
+                binding.setProperty('password', null)
+            }
+         })
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> return 0 })
+
+        nullScript.commonPipelineEnvironment.configuration = [steps:
+                                     [transportRequestUploadFile:
+                                         [
+                                          cmCredentialsId: 'CM',
+                                          cmEndpoint: 'https://example.org/cm'
+                                         ]
+                                     ]
+                                 ]
+    }
+
+    @Test
+    public void changeIdNotProvidedTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("Change id not provided (parameter: 'changeId').")
+
+        jsr.step.call(script: nullScript, transportRequestId: '001', applicationId: 'app', filePath: '/path')
+    }
+
+    @Test
+    public void transportRequestIdNotProvidedTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("Transport Request id not provided (parameter: 'transportRequestId').")
+
+        jsr.step.call(script: nullScript, changeId: '001', applicationId: 'app', filePath: '/path')
+    }
+
+    @Test
+    public void applicationIdNotProvidedTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("Application id not provided (parameter: 'applicationId').")
+
+        jsr.step.call(script: nullScript, changeId: '001', transportRequestId: '001', filePath: '/path')
+    }
+
+    @Test
+    public void filePathNotProvidedTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("File path not provided (parameter: 'filePath').")
+
+        jsr.step.call(script: nullScript, changeId: '001', transportRequestId: '001', applicationId: 'app')
+    }
+
+    @Test
+    public void uploadFileToTransportRequestFailureTest() {
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> return 1 })
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("Cannot upload file '/path' for change document '001' with transport request '001'. Return code from cmclient: 1.")
+
+        jsr.step.call(script: nullScript, changeId: '001', transportRequestId: '001', applicationId: 'app', filePath: '/path')
+    }
+
+    @Test
+    public void uploadFileToTransportRequestSuccessTest() {
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> return 0 })
+
+        jsr.step.call(script: nullScript, changeId: '001', transportRequestId: '001', applicationId: 'app', filePath: '/path')
+
+        assert jlr.log.contains("[INFO] Uploading file '/path' to transport request '001' of change document '001'.")
```

Comment:
See above: also assert the command line or even better that the corresponding ChangeManagement function has been called.

Reply:
Yes, that should be tested in the ChangeManagementTest.class.